src/Simple.Migrations/DatabaseProvider/OracleDatabaseProvider.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Data.Common;
+
+namespace SimpleMigrations.DatabaseProvider
+{
+    /// <summary>
+    /// Class which can read from / write to a version table in an PostgreSQL database
+    /// </summary>
+    /// <remarks>
+    /// PostgreSQL supports advisory locks, so these are used to guard against concurrent migrators.
+    /// </remarks>
+    public class OracleDatabaseProvider : DatabaseProviderBaseWithVersionTableLock
+    {
+        /// <summary>
+        /// Name of Oracle Sequence for generating Id in version table
+        /// 
+        /// </summary>
+        public string VersionSequenceName { get; set; } = "SEQ_VERSION_TABLE";
+        
+        /// <summary>
+        /// Controls whether or not to try and create the schema if it does not exist.
+        /// </summary>
+        /// <remarks>
+        /// If this is set to false then no schema is created. It is the user's responsibility to create the schema
+        /// (if necessary) with the correct name and permissions before running the <see cref="SimpleMigrator"/>. This may be
+        /// required if the user which Simple.Migrator is running as does not have the correct permissions to check whether the
+        /// schema has been created.
+        /// </remarks>
+        public bool CreateSchema { get; set; } = false;
+        
+
+        /// <summary>
+        /// Initialises a new instance of the <see cref="OracleDatabaseProvider"/> class
+        /// </summary>
+        /// <param name="connectionFactory">Connection to use to run migrations. The caller is responsible for closing this.</param>
+        public OracleDatabaseProvider(Func<DbConnection> connectionFactory)
+            : base(connectionFactory)
+        {
+            TableName = "VERSION_INFO";
```

Comment:
Is using all caps a particularly strong oracle-ism? I.e. is there a compelling reason to move away from the name used by other database providers?

Reply:
Yes, it is standard, oracle names are by default case insensitive and even if you will use small caps it will be stored in dictionary as all caps